src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -748,6 +752,35 @@ class ScriptRegistrySwapHooks<SceneScript, Base>
     bool swapped;
 };
 
+/// This hook is responsible for swapping QuestScript's
```

Comment:
can you explain when we need these ScriptRegistrySwapHooks? i was thinking that we don't need such hooks for not "living" objects and scripts not bound to any object

Reply:
Traesh, can you respond to this?

Reply:
I think joschiwald is right, maybe @Naios could confirm

Reply:
Yes, the hook isn't needed here.
It's only needed for scripts with a custom unload logic such as database bound scripts.

Reply:
Well this script is database bound, it's just not entity linked

Reply:
@Traesh Yes, you are right, my bad. The code is fine.
I was just wondering the fact that you only implemented the assertions,
but this is equal to the hooks of `InstanceMapScript` and maybe those could be merged in the future.
:+1: